app/frontend/src/features/communities/discussions/DiscussionPage.test.tsx

```diff
@@ -0,0 +1,108 @@
+import {
+  render,
+  screen,
+  waitForElementToBeRemoved,
+} from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { Route, Switch } from "react-router-dom";
+import { discussionBaseRoute, discussionRoute } from "routes";
+import { service } from "service";
+import discussions from "test/fixtures/discussions.json";
+import threadRes from "test/fixtures/getThreadRes.json";
+import { getHookWrapperWithClient } from "test/hookWrapper";
+import { getUser } from "test/serviceMockDefaults";
+import { assertErrorAlert, mockConsoleError, MockedService } from "test/utils";
+
+import { PREVIOUS_PAGE } from "../constants";
+import DiscussionPage, { CREATOR_LOADING_TEST_ID } from "./DiscussionPage";
+
+const getUserMock = service.user.getUser as MockedService<
+  typeof service.user.getUser
+>;
+const getDiscussionMock = service.discussions.getDiscussion as MockedService<
+  typeof service.discussions.getDiscussion
+>;
+const getThreadMock = service.threads.getThread as MockedService<
+  typeof service.threads.getThread
+>;
+
+function renderDiscussion() {
+  const { client, wrapper } = getHookWrapperWithClient({
+    initialRouterEntries: [
+      "/previous-page",
+      `${discussionBaseRoute}/1/what-is-there-to-do-in-amsterdam`,
+    ],
+    initialIndex: 1,
+  });
+  render(
+    <Switch>
+      <Route exact path={discussionRoute}>
+        <DiscussionPage />
+      </Route>
+      <Route exact path="/previous-page">
+        <h1 data-testid="previous-page">Previous page</h1>
+      </Route>
+    </Switch>,
+    { wrapper }
+  );
+
+  return client;
+}
+
+describe("Discussion page", () => {
+  beforeEach(() => {
+    getUserMock.mockImplementation(getUser);
+    getDiscussionMock.mockResolvedValue(discussions[0]);
+    getThreadMock.mockResolvedValue(threadRes);
+  });
+
+  it("renders the discussion successfully", async () => {
+    renderDiscussion();
+
+    await waitForElementToBeRemoved(screen.getByRole("progressbar"));
+
+    // Author and discussion content assertions
+    expect(
+      screen.getByRole("heading", {
+        level: 1,
+        name: "What is there to do in Amsterdam?",
+      })
+    ).toBeVisible();
+    expect(
+      screen.getByText(/i'm looking for activities to do here!/i)
+    ).toBeVisible();
+    expect(screen.getByText("Added by:")).toBeVisible();
+    expect(screen.getByText("Funny Cat current User")).toBeVisible();
+    expect(screen.getByText("Created at Jan 01, 2020")).toBeVisible();
+  });
+
+  it("renders a loading skeleton if the user info is still loading", async () => {
+    getUserMock.mockImplementation(async () => new Promise(() => undefined));
+    renderDiscussion();
+    await waitForElementToBeRemoved(screen.getByRole("progressbar"));
+
+    expect(await screen.findByTestId(CREATOR_LOADING_TEST_ID)).toBeVisible();
```

Comment:
Can we avoid the test id by checking that the discussion info is shown, user info isn't shown and there is a progress bar? Or is that too inaccurate?

Reply:
Checking discussion is shown but user is not shown could work actually. If it errors, it'd not be loading and falls back to "Unknown user" instead since 1) I can't think of a good way to show an error state here without it looking terrible and 2) this could error if in the future the user deactivates/deletes their account, so falling back to "Unknown user" seems reasonable to me.

The progress bar bit is quite impossible because the `Skeleton` component doesn't actually have a `progressbar` role and MUI implements it with a span... so that's why I added a test ID as spans are so generic there's nothing else in the query hierarchy we can use to select this properly.